baseapp/internal/protocompat/protocompat.go

```diff
@@ -206,3 +206,30 @@ func isProtov2(md grpc.MethodDesc) (isV2Type bool, err error) {
 	_, _ = md.Handler(nil, nil, pullRequestType, doNotExecute)
 	return
 }
+
+// MessageName returns the fully-qualified name of the message.
+// Supports proto2 and gogoproto.
+func MessageName(msg protoiface.MessageV1) string {
```

Comment:
This does just that right: https://github.com/cosmos/cosmos-sdk/blob/main/codec/types/util.go#L8-L15

Reply:
good catch